### DISCUSSION TITLE
Support for SR Linux EVPN eBGP Unnumbered

### DISCUSSION THREAD

**Discussion by hellt** [ipspace/netlab#1295] [Features]
Hi @ipspace , all,

I wanted to revisit support for EVPN eBGP Unnumbered support on SR Linux.

After doing this:

```
[root@clab X]# netlab up -p clab -d srlinux ../tests/integration/evpn/15-vxlan-ebgp-unnumbered.yml
 
┌──────────────────────────────────────────────────────────────────────────────────┐
│ CREATING configuration files                                                     │
└──────────────────────────────────────────────────────────────────────────────────┘
[ERRORS]  Errors found in 15-vxlan-ebgp-unnumbered.yml
[TYPE]    quirks: SR Linux (spine) does not support EVPN AF over interface EBGP sessions
[DATA]    Neighbor l1 interface ethernet-1/1
[TYPE]    quirks: SR Linux (spine) does not support EVPN AF over interface EBGP sessions
[DATA]    Neighbor l2 interface ethernet-1/2
[TYPE]    quirks: SR Linux (l1) does not support EVPN AF over interface EBGP sessions
[DATA]    Neighbor spine interface ethernet-1/1
[FATAL]   Cannot proceed beyond this point due to errors, exiting
[root@clab X]#
```

how can I make netlab bow down and generate the config to the best of its ability to investigate which config is in place (if any)

**Comment by ipspace** [comment]
You have to remove the "this does not work" check in netsim/devices/srlinux.py

https://github.com/ipspace/netlab/blob/dev/netsim/devices/srlinux.py#L42

That check was originally in a Jinja2 template, I moved it to device quirks to ensure the lab won't start (better than crashing an Ansible playbook in an obscure way)

**Reply by hellt** [reply]
merci, will fiddle with this more, since we do support this flavor of ebgp unnumbered, I will just need to untangle the template sequencing to understand where to add what

**Reply by ipspace** [reply]
The best thing to do (after starting the lab and fiddling with the config to make it work) would be to use **netlab initial -o** to generate config snippets in the `config` directory. That will tell you which template generates what part of configuration.

Obviously, you could also tell me what to fix. That might be faster ;)

**Reply by jbemmel** [reply]
> merci, will fiddle with this more, since we do support this flavor of ebgp unnumbered, I will just need to untangle the template sequencing to understand where to add what

It wasn't supported at the time. Simply remove those lines in 'quirks', and bump the default SR Linux container version to the latest one

**Reply by ipspace** [reply]
As @hellt already knows, the 'latest one' (24.7) doesn't work because the BGP data model changed **import-policy** and **export-policy** from **leafref** to **leafref-list**, resulting in breaking BGP template. I will fix that after 24.10 comes out (and move to 24.10)

**Reply by hellt** [reply]
@ipspace if netlab' strategy would be to support only LTS releases, then it would be fair to mention the release version in the table with the results. This would make it clear what versions have been tested without ambiguity

**Reply by ipspace** [reply]
Do I really have to have that same discussion again? _netlab_ strategy is not to "support only LTS releases" but to have a stable product that works while minimizing wasted efforts (because we're not on vendor payroll). It's not my problem if you guys decided it makes sense to run fast and break things. SR Linux is the only OS so far that broke something fundamental (BGP routing policies) in such a blatant way.

The only other case (IIRC) where we had to fix netlab templates was some change in FRRouting OSPF setup, but even that was probably introduced, then the old thing deprecated, then the old thing disappeared, and I just didn't realize that for a while.

Anyhow, I agree it would make sense to publish the list of device images used in the tests. Will add to the top-level page, probably with the next _netlab_ release.

**Reply by hellt** [reply]
In our email thread, I, who is on a vendor payroll, said that I am willing to try and adapt the templates to the recent version to support the use cases marked as "failed/config failed".  
Which is another way to say that I was not expecting you to fix config changes that happened between the releases.

If that was not said clearly, the above is a reiteration.

You proposed waiting for an LTS release anyway. I am fine with that, no problem.

**Reply by ipspace** [reply]
> In our email thread, I, who is on a vendor payroll, said that I am willing to try and adapt the templates to the recent version to support the use cases marked as "failed/config failed".

That is correct. 

Let's go. I created the "srl_247" branch in which I made the minimum changes to make BGP work on SR Linux 24.7. I also created an "enhancement" issue (#1298) documenting what else is failing and why. It would be great if you could figure out how to fix the problems and submit a PR against the "srl_247" branch. After the integration tests pass, I'll merge the srl_247 branch with the dev branch.

**Comment by ipspace** [comment]
Integrated in #1300 ;)
